tests/core/full_node/full_sync/test_full_sync.py

```diff
@@ -412,3 +415,59 @@ async def test_sync_none_wp_response_backward_comp(self, three_nodes, default_10
         assert res is None
         duration = time.time() - start
         assert duration > 5
+
+    @pytest.mark.asyncio
+    async def test_bad_peak_cache(
+        self, three_nodes, default_400_blocks, blockchain_constants, self_hostname, consensus_mode, bt
+    ):
+        if consensus_mode != Mode.PLAIN:
+            pytest.skip("Skipped test")
+        full_node_1, full_node_2, full_node_3 = three_nodes
+        bt.constants = blockchain_constants.replace(SOFT_FORK4_HEIGHT=1000000)
```

Comment:
suggestions ? 
this is the only way i managed to make it work inside the mess of fork flags and diff confs for each flag

Reply:
I guess maybe `get_b_tools()` or `get_b_tools_1()`.  Or possibly calling `create_block_tools()` or `create_block_tools_async()` directly.  I'm not particularly familiar with the details here beyond feeling that we need to make the global/session-scoped nature more clear when reading the code and also make it more obvious what else to use.

Reply:
can you explain what you are worried about with it being scooped i prefer getting this right if its not to much changes

Reply:
since bt is session-scoped if you change something, that change I think gets propagated to the next test using that fixture. Which might not be desired.

Reply:
Here's a way to get around the BlockTools issue without getting into the weeds.
Replace the node setup above with:

```
@pytest_asyncio.fixture(scope="function")
async def my_node() -> AsyncIterator[Tuple[List[Service], List[FullNodeSimulator], BlockTools]]:
    async for _ in setup_simulators_and_wallets_service(1, 0, {SOFT_FORK4_HEIGHT: 1000000}):
        yield _

# Then, create each full node with a specific bt. eg if full_node_1 is the special one:

full_node_1 = my_node
full_node_2, full_node_3 = two_nodes
```

It sounds like we will get some time to undo some of this test jank soon.